embulk-core/src/main/java/org/embulk/deps/EmbulkSelfContainedJarFiles.java

```diff
@@ -128,5 +142,9 @@ static Resource getSingleResource(final String targetResourceName) {
         return list;
     }
 
-    private static final ArrayList<String> JAR_RESOURCE_NAMES = new ArrayList<>();
+    // Category for embulk-core's own self-contained dependencies.
+    public static final String CORE = "";
```

Comment:
Safer to give it some real value to avoid passing empty strings by mistake in future?

Reply:
Also a reasonable point. I at first chose an empty string so that it will never conflict with given categories accidentally (expected to be an embedded plugin name: such as `embulk-parser-csv` as of now (*1)), but yeah, mistakes can easily bring an empty string there.
(*1): https://github.com/embulk/embulk/pull/1360/files#diff-02ae5d423faebe1c8fd3e17f337912a9b5e29157d471a7603cd7053c2e587b72R79

In 33f9d8bf622a72b6762a0f6d736b91f32b67d9b2, set it as `"$embulk-deps$"` which explicitly seems like a special thing.